log/log.go

```diff
@@ -107,3 +133,19 @@ func getConsoleEncoder() zapcore.Encoder {
 
 	return zapcore.NewConsoleEncoder(encoderConfig)
 }
+
+type ctxLoggerKey string
+
+const ctxLogger ctxLoggerKey = "logger"
+
+// ToContext allows setting the logger on the context
+func ToContext(ctx context.Context, l Logger) context.Context {
+	return context.WithValue(ctx, ctxLogger, l)
+}
+
+// FromContext returns the logger from the context when set with CtxWithLogger.
+// If not found, it returns a nil value.
+func FromContext(ctx context.Context) (Logger, bool) {
+	l, ok := ctx.Value(ctxLogger).(Logger)
+	return l, ok
```

Comment:
Should this be "FromContextOrDefault" and handled the error logging and fallback here instead of whenever it is used?

Reply:
That's a good idea, I like it! Will refactor it.